meilisearch-http/src/option.rs

```diff
@@ -261,33 +263,33 @@ impl Opt {
-        {
-            match std::fs::read(&config_file_path) {
-                Ok(config) => {
-                    // If the file is successfully read, we deserialize it with `toml`.
-                    let opt_from_config = toml::from_slice::<Opt>(&config)?;
-                    // Return an error if config file contains 'config_file_path'
-                    // Using that key in the config file doesn't make sense bc it creates a logical loop (config file referencing itself)
-                    if opt_from_config.config_file_path.is_some() {
-                        anyhow::bail!("`config_file_path` is not supported in config file")
-                    }
-                    // We inject the values from the toml in the corresponding env vars if needs be. Doing so, we respect the priority toml < env vars < cli args.
-                    opt_from_config.export_to_env();
-                    // Once injected we parse the cli args once again to take the new env vars into scope.
-                    opts = Opt::parse();
-                    config_read_from = Some(config_file_path);
+            .or_else(|| env::var("MEILI_CONFIG_FILE_PATH").map(PathBuf::from).ok())
+            .unwrap_or_else(|| PathBuf::from(DEFAULT_CONFIG_FILE_PATH));
+
+        match std::fs::read(&config_file_path) {
+            Ok(config) => {
+                // If the file is successfully read, we deserialize it with `toml`.
+                let opt_from_config = toml::from_slice::<Opt>(&config)?;
+                // Return an error if config file contains 'config_file_path'
+                // Using that key in the config file doesn't make sense bc it creates a logical loop (config file referencing itself)
+                if opt_from_config.config_file_path.is_some() {
+                    anyhow::bail!("`config_file_path` is not supported in config file")
                 }
-                // If we have an error while reading the file defined by the user.
-                Err(_) if opts.config_file_path.is_some() => anyhow::bail!(
-                    "unable to open or read the {:?} configuration file.",
-                    opts.config_file_path.unwrap().display().to_string()
-                ),
-                _ => (),
+                // We inject the values from the toml in the corresponding env vars if needs be. Doing so, we respect the priority toml < env vars < cli args.
+                opt_from_config.export_to_env();
+                // Once injected we parse the cli args once again to take the new env vars into scope.
+                opts = Opt::parse();
+                config_read_from = Some(config_file_path);
             }
+            // If we have an error while reading the file defined by the user.
+            Err(_) if opts.config_file_path.is_some() => anyhow::bail!(
+                "unable to open or read the {:?} configuration file.",
+                opts.config_file_path.unwrap().display().to_string()
+            ),
```

Comment:
It would be great if you display the originating error in the message, please.
```suggestion
            Err(e) if opts.config_file_path.is_some() => anyhow::bail!(
                "unable to open or read the {:?} configuration file: {}.",
                opts.config_file_path.unwrap().display().to_string(), e,
            ),
```

Reply:
Fixed in d301563